bindings/java/src/org/sleuthkit/datamodel/CaseDatabaseFactory.java

```diff
@@ -436,25 +436,26 @@ private void createAccountTables(Statement stmt) throws SQLException {
 		
 		// References tsk_hosts
 		stmt.execute("CREATE TABLE tsk_os_account_realms (id " + dbQueryHelper.getPrimaryKey() + " PRIMARY KEY, "
-				+ "name TEXT NOT NULL, "	// realm name - host name or domain name
+				+ "realm_name TEXT DEFAULT NULL, "	// realm name - for a domain realm, may be null
 				+ "realm_addr TEXT DEFAULT NULL, "		// a sid/uid or some some other identifier, may be null
-				+ "host_id " + dbQueryHelper.getBigIntType() + " DEFAULT NULL, " // if the realm just comprises of a single host
-				+ "name_type INTEGER, "	// indicates if the realm name was  was expressed or inferred 
-				+ "UNIQUE(name, host_id), "
-				+ "FOREIGN KEY(host_id) REFERENCES tsk_hosts(id) )");
+				+ "realm_signature TEXT NOT NULL, "	// is either the realm_addr or the realm_addr, one of them must be known.  This is used to prevent duplicates.
+				+ "scope_host_id " + dbQueryHelper.getBigIntType() + " DEFAULT NULL, " // if the realm scope is a single host
+				+ "scope_confidence INTEGER, "	// indicates whether we know for sure the realm scope or if we are inferring it
+				+ "UNIQUE(realm_signature, scope_host_id), "
```

Comment:
@raman-bt and @bcarrier, I think we have a problem with this UNIQUE constraint. At least for SQLite, NULL values in the same column are not equivalent. See sqlitetutorial.net/sqlite-unique-constraint/. So these two rows would be possible, even with this constraint:
S-1-5-21, NULL
S-1-5-21, NULL

Reply:
Addressed uniqueness by making a signature string as follows:
(realmAddr|realmName)_(hostId|"DOMIAN")

Fixed the UNIQUE constraint on table to be only on signature column.